server.go

```diff
@@ -190,13 +281,16 @@ func (srv *Server) run() {
 			if srv.ReplayAll || len(sub.lastEventID) > 0 {
 				repo, ok := repos[sub.channel]
 				if ok {
-					go replay(repo, sub)
+					batchCh := repo.Replay(sub.channel, sub.lastEventID)
```

Comment:
Again, in the interest of summarizing the important bits of this PR to be sure I understand them, the old code used to spawn a new goroutine that actually wrote individual messages to the subscription, as it replayed the messages.

The new code sends a channel that can be used to read the replayable messages, and passes that to the main goroutine that writes to this subscription (which I summarized in a comment above). Thus, there is only one goroutine sending messages to the subscription.

Reply:
That's right.